### PR TITLE
Forgotten password

### PR DESCRIPTION
**Pull request by GuenoleCHERUBIN**
Add Forgotten Password

Added functions in siteController that manage
the password forgotten.
Added the forms and views that display this
option for the customer.

### CODE REVIEW

File: controllers/SiteController.php [hunk 1]
```diff
@@ -167,4 +172,132 @@ private function checkIfUserMatchToToken($token)
...
+     */
+    private function mailToUserPasswordForgotten(User $user)
+    {
+        $mail = Util::getConfiguredMailerForMailhog();
+        $mail->addAddress($user->email);
+        $mail->isHTML(false);
+        $mail->CharSet = 'UTF-8';
+
+        $mail->Subject = "Forgotten your Collab'media account password";
+        $mail->Body = 'Click on the following link to reset your password : http://127.0.0.1:8000/site/change-password/' . $user->token ;
```
Comment (EmileCalixte) [inline controllers/SiteController.php]: What will happen when we will run the app on our final domain ? What will happen if I have Docker Toolbox and I the IP address of my project is not localhost ?

Please create a public const 'APP_BASE_URL' in Util component, and use this constant everywhere you need an absolute URL.

File: models/forms/ChangePasswordForm.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+class ChangePasswordForm extends Model
+{
+    public $password;
+    public $confirmPassword;
+
+    public function rules()
+    {
+        return [
+            // username and password are both required
+            [['password', 'confirmPassword'], 'required'],
```
Comment (EmileCalixte) [inline models/forms/ChangePasswordForm.php:21]: The comment does not match the line below

File: models/forms/ForgottenPasswordForm.php [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+use yii\base\Model;
+class ForgottenPasswordForm extends Model
+{
+    public $email;
+
+    public function rules()
+    {
+        return [
+            // username and password are both required
+            [['email'], 'required'],
```
Comment (EmileCalixte) [inline models/forms/ForgottenPasswordForm.php:13]: Same

**Review by EmileCalixte — CHANGES_REQUESTED** [review]

File: controllers/SiteController.php (hunk 1 above)
Comment (alexisljn) [reply controllers/SiteController.php]: I fully agree

File: views/site/login.php [hunk 4]
```diff
@@ -24,10 +24,14 @@
         ],
     ]); ?>
 
+
```
Comment (alexisljn) [inline views/site/login.php]: Why those spaces ?

**Review by alexisljn — CHANGES_REQUESTED** [review]

File: views/site/login.php (hunk 4 above)
Comment (GuenoleCHERUBIN) [reply views/site/login.php]: An error from myself, i had placed the password forgotten link there.

**Review by EmileCalixte — APPROVED** [review]

**Review by alexisljn — APPROVED** [review]

**Pull request merged**